### PR TITLE
Add Router 3 into Main Repo

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: scripts/ci-lite/build.sh [hunk 1]
```diff
@@ -13,5 +13,6 @@ $(npm bin)/tsc -v
 $(npm bin)/tsc -p tools
 cp tools/@angular/tsc-wrapped/package.json dist/tools/@angular/tsc-wrapped
 node dist/tools/@angular/tsc-wrapped/src/main -p modules
+node dist/tools/@angular/tsc-wrapped/src/main -p modules/@angular/router
```
Comment (vicb) [inline scripts/ci-lite/build.sh]: do you need this ?

File: modules/@angular/router/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License
```
Comment (vicb) [inline modules/@angular/router/LICENSE]: remove ?

File: modules/@angular/router/src/router_providers.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *   {path: '/home', component: Home}
+ * ];
+ *
+ * bootstrap(AppCmp, [provideRouter(router)]);
+ * ```
+ */
+export function provideRouter(config: RouterConfig, opts: common.ExtraOptions = {}): any[] {
+  return [
+    {provide: PlatformLocation, useClass: BrowserPlatformLocation},
+    ...common.provideRouter(config, opts)
```
Comment (vicb) [inline modules/@angular/router/src/router_providers.ts]: `...` could be removed ?

File: scripts/ci-lite/build.sh (hunk 1 above)
Comment (vsavkin) [inline scripts/ci-lite/build.sh]: We do. We build and test the router module separately.

File: modules/@angular/router/src/router_providers.ts (hunk 3 above)
Comment (vsavkin) [inline modules/@angular/router/src/router_providers.ts]: We can. This wasn't introduce in this PR though.

**Comment by vicb** [comment]
merged at 15911367a2573b46a54dffd12b2aa841f01c08bd

**Pull request closed without merging**